rdtools/analysis_chains.py

```diff
@@ -0,0 +1,829 @@
+        `temperature_ambient_clearsky` are not passed, they will be modeled
+        based on location and orientation.
+
+        Parameters
+        ----------
+        pvlib_location : pvlib.location.Location
+            Used for calculating clearsky temperature and irradiance
+        pv_azimuth : numeric
+            Azimuth of PV array in degrees from north
+        pv_tilt : numeric
+            Tilt of PV array in degrees from horizontal
+        poa_global_clearsky : pd.Series
+            Right-labeled time Series of clear-sky plane of array irradiance
+        temperature_cell_clearsky : pd.Series
+            Right-labeled time series of cell temperature in clear-sky conditions
+            in Celsius. In practice, back of module temperature works as a good
+            approximation.
+        temperature_ambient_clearsky : pd.Series
+            Right-label time series of ambient temperature in clear sky conditions
+            in Celsius
+        albedo : numeric
+            Albedo to be used in irradiance transposition calculations
+
+        '''
+        interp_freq = self.interp_freq
+        max_timedelta = self.max_timedelta
+
+        if interp_freq is not None:
+            if poa_global_clearsky is not None:
+                poa_global_clearsky = normalization.interpolate(
+                    poa_global_clearsky, interp_freq, max_timedelta)
+            if temperature_cell_clearsky is not None:
+                temperature_cell_clearsky = normalization.interpolate(
+                    temperature_cell_clearsky, interp_freq, max_timedelta)
+            if temperature_ambient_clearsky is not None:
+                temperature_ambient_clearsky = normalization.interpolate(
+                    temperature_ambient_clearsky, interp_freq, max_timedelta)
+            if isinstance(pv_azimuth, (pd.Series, pd.DataFrame)):
+                pv_azimuth = normalization.interpolate(
+                    pv_azimuth, interp_freq, max_timedelta)
+            if isinstance(pv_tilt, (pd.Series, pd.DataFrame)):
+                pv_tilt = normalization.interpolate(
+                    pv_tilt, interp_freq, max_timedelta)
+
+        self.pvlib_location = pvlib_location
+        self.pv_azimuth = pv_azimuth
+        self.pv_tilt = pv_tilt
+        self.poa_global_clearsky = poa_global_clearsky
+        self.temperature_cell_clearsky = temperature_cell_clearsky
+        self.temperature_ambient_clearsky = temperature_ambient_clearsky
+        self.albedo = albedo
+
+    def calc_clearsky_poa(self, times=None, rescale=True, **kwargs):
+        '''
+        Calculate clearsky plane-of-array irradiance and stores in self.poa_global_clearsky
+
+        Parameters
+        ----------
+        times : pandas.DateTimeIndex
+            times on for which to calculate clearsky poa
+        rescale : bool
+            Whether to attempt to rescale clearsky irradiance to measured
+        model : str
```

Comment:
Don't think this actually exists.  The model can be passed through via the `**kwargs`, so I think it's ok to just delete this docstring parameter entry.